### PR TITLE
Socks upgrade

### PR DESCRIPTION
**Pull request by mike820324**
This PR upgrade socks5 to v 0.2.1.

- upgrade construct to 2.8.8
- modify pyca_tls
- upgrade socks5 to v0.2.1
- fix socks related tests
- add test for construct and protocol.tls.TlsClientHello

### CODE REVIEW

**Comment by mike820324** [comment]
Hi @chhsiao90,
help me review this, tks.

File: microproxy/test/layer/test_socks.py [hunk 1]
```diff
@@ -118,7 +112,7 @@ def test_socks_request_ipv4(self):
 
         self.assertIsNotNone(dest_stream)
         self.assertFalse(dest_stream.closed())
-        self.assertEqual(host, "127.0.0.1")
+        self.assertEqual(str(host), "127.0.0.1")
```
Comment (chhsiao90) [inline microproxy/test/layer/test_socks.py]: I think ```self.assertEqual(host, u"127.0.0.1")``` would better if host type is unicode.
That it's more clearly to know what's the type of host.

File: microproxy/test/layer/test_socks.py [hunk 2]
```diff
@@ -194,7 +180,7 @@ def test_handle_connection_timeout(self):
         self.assertIsInstance(self.event, Response)
         self.assertEqual(self.event.status, RESP_STATUS["NETWORK_UNREACHABLE"])
         self.assertEqual(self.event.atyp, ADDR_TYPE["IPV4"])
-        self.assertEqual(self.event.addr, "1.2.3.4")
+        self.assertEqual(str(self.event.addr), "1.2.3.4")
```
Comment (chhsiao90) [inline microproxy/test/layer/test_socks.py]: Same problem, cast the variable that we want to measure make it's unknown that what's the actual type of the variable.

**Review by chhsiao90 — CHANGES_REQUESTED** [review]
LGTM, nice work~
Just had some comments~

**Comment by chhsiao90** [comment]
Wow, the option ```Changes requested``` of code review is cool~

**Comment by mike820324** [comment]
Will do, I'll push a new commit later.

**Comment by mike820324** [comment]
Thanks for the review.
The new code review functionality is really good, except I forgot to press submit. Orz

**Review by chhsiao90 — APPROVED** [review]

**Comment by mike820324** [comment]
Fix based on review, thanks.

**Review by chhsiao90 — APPROVED** [review]
LGTM~

**Pull request merged**